client/blocks/upsell-nudge/index.jsx

```diff
@@ -56,11 +56,9 @@ export const UpsellNudge = ( {
 		return null;
 	}
 
-	if ( ! href && site ) {
-		href = addQueryArgs( { feature, plan }, `/plans/${ site.slug }` );
-	}
+	const link = href ? href : addQueryArgs( { feature, plan }, `/plans/${ site.slug }` );
```

Comment:
I think there is still an error. Before setting a link to `/plans/:site` we need to make sure `site` is defined. Probably it's better to revert all these changes at all.

Reply:
I think that this can't be reverted, else nudges such as the "Remove the footer credit entirely" one won't work since they don't contain this link. But I don't think there can be a case where `site` is undefined.

https://github.com/Automattic/wp-calypso/blob/ff1fdca234890a0d8dbb95df0e7ce3345fb2e50c/client/blocks/upsell-nudge/index.jsx#L46

Reply:
> I think that this can't be reverted, else nudges such as the "Remove the footer credit entirely" one won't work since they don't contain this link

I think they work currently. At least they work for me when testing in production:

<img width="1025" alt="Screen Shot 2020-04-16 at 17 17 52" src="https://user-images.githubusercontent.com/1233880/79474048-3aa0d780-8006-11ea-8c82-570116a9360c.png">

Which makes with what I see with the current code: we add a link when none is provided and there is a selected site (`! href && site`).

> But I don't think there can be a case where site is undefined.

When `forceDisplay` is true.

Reply:
Hmm, this confuses me. I say the same thing on production, but when I remove this line and keep `return <Banner { ...props } showIcon={ showIcon } className={ classes } href={ href } />;`, I see this:

<img width="751" alt="Screenshot 2020-04-16 at 16 25 30" src="https://user-images.githubusercontent.com/43215253/79474968-ff4eda80-7ffe-11ea-9754-11c3cbffd094.png">

Reply:
When you say you remove this line, you mean this one?

```
const link = href ? href : addQueryArgs( { feature, plan }, `/plans/${ site.slug }` );
``` 

I guess you do, but just to confirm, you also add the previous code when removing that line, right?

```
if ( ! href && site ) {
	href = addQueryArgs( { feature, plan }, `/plans/${ site.slug }` );
}
```

Reply:
That explains it, oops! Fixed!